src/server/trpc/router/admin/events/create.ts

```diff
@@ -10,21 +13,39 @@ export const createEvent = protectedProcedure
       endDate: z.date(),
       departments: z.array(z.string()),
       attendees: z.array(z.string()),
+      isQrRequired: z.boolean(),
     })
   )
-  .mutation(async ({ input }) => {
-    const prisma = new PrismaClient()
+  .mutation(async ({ ctx, input }) => {
+    try {
+      let qr_code: string | undefined
+      const id = randomUUID()
 
-    await prisma.event.create({
-      data: {
-        attendees: {
-          connect: input.attendees.map((attendee) => {
-            return { id: attendee }
-          }),
+      if (input.isQrRequired) {
```

Comment:
Feel like in this block we can try and add in a log just in case we need to verify the link that the qr code generates and other details